### PR TITLE
Better error messages in async

### PR DESCRIPTION
**Pull request by clux**
A couple of quick bad errors messages i found via filebacked and helm module

### CODE REVIEW

File: shipcat_cli/src/helm.rs [hunk 1]
```diff
@@ -74,19 +74,24 @@ pub async fn template(mf: &Manifest, output: Option<PathBuf>) -> Result<String>
...
+        f.sync_data().await?;
         debug!(
             "Wrote helm template for {} to {}: \n{}",
             mf.name,
             pth.display(),
             tpl
         );
     };
-    fs::remove_file(hfile)?;
+    if output.is_some() {
```
Comment (janslow) [inline shipcat_cli/src/helm.rs]: This can be merged with the above `if` statement

File: shipcat_filebacked/src/load.rs [hunk 2]
```diff
@@ -83,10 +83,14 @@ impl ManifestSource {
...
         let mut all = vec![];
         for service in Self::all_names() {
-            let source_path = Self::services_dir().join(service).join("manifest.yml");
+            let source_path = Self::services_dir().join(&service).join("manifest.yml");
             debug!("Loading service manifest from {:?}", source_path);
-            let source: ManifestSource = read_from(&source_path).await?;
-            let manifest = source.build_base(conf)?;
+            let source: ManifestSource = read_from(&source_path)
+                .await
+                .chain_err(|| ErrorKind::InvalidManifest(service.clone()))?;
```
Comment (janslow) [inline shipcat_filebacked/src/load.rs]: Does this keep the original error details as well?

File: shipcat_filebacked/src/load.rs [hunk 3]
```diff
@@ -155,7 +161,10 @@ async fn read_from<T: DeserializeOwned>(path: &PathBuf) -> Result<T> {
     if data.is_empty() {
         bail!("Manifest file {} is empty", path.display());
     }
-    Ok(serde_yaml::from_str(&data)?)
+    match serde_yaml::from_str(&data) {
+        Err(e) => bail!("Manifest file {} did not parse: {}", path.display(), e),
```
Comment (janslow) [inline shipcat_filebacked/src/load.rs]: Could you say that it's invalid YAML (as that's useful for debugging, as you know to look for a basic syntax error)?

e.g., 

```suggestion
        Err(e) => bail!("Manifest file {} did not parse as YAML: {}", path.display(), e),
```

File: shipcat_filebacked/src/load.rs (hunk 2 above)
Comment (clux) [reply shipcat_filebacked/src/load.rs]: Yeah, this simply pushes the InvalidManifest in front of the previous stack in error-chain

**Pull request merged**